### ISSUE TITLE
ma.log waiting for low priority thread should report number of threads left

### ISSUE THREAD

**Issue by resotek** [infiniteautomation/ma-core-public#611]
**Problem:** Sometimes during shutdown the ma.log reports that mango is waiting for low priority threads to complete. This can take a while. The problem is that there is no indication as to how much progress, if any, is being made to that end.

**Solution**: ma.log should log the number of threads left to complete, so the user can determine whether to wait or kill the process.

**Comment by terrypacker** [comment]
There is only 1 low priority thread in Mango but it has an unbounded queue of tasks it must process.  New logging messages have been added to show the size of the queue.  Mango should never take more than 6 minutes to terminate all 3 of the high, medium and low priority tasks.  

This could be made adjustable...

**Comment by terrypacker** [comment]
Make the time to shutdown configurable via env.properties.

**Comment by terrypacker** [comment]
Perhaps we want a shorter default shutdown time than 6 min + 6 min = 12 minutes?

**Comment by Puckfist** [comment]
The comments in the env.properties file do not reflect the reality. It says that each tick is 10 seconds, but it is clearly 5 in BackgroundProcessing lines 381 && 383. I suppose that's 10 seconds if you have tasks in both categories to wait for? Mine was only waiting 5 when I breakpoint'ed an email handler on shutdown.

Also, the log statement in Lifecycle:469 says waiting for high and low priority tasks, but it's really only waiting on the high priority tasks, right?

**Comment by Puckfist** [comment]
Seems like the log messages would still be misleading, since it says " waiting " + rewaits + " more seconds" but it is possible to sleep for 2 seconds each decrement of rewaits if low priority tasks also need termination.

**Comment by terrypacker** [comment]
Tested and working as advertised.
